packages/http-client-csharp/emitter/src/emitter.ts

```diff
@@ -78,13 +77,7 @@ export async function $onEmit(context: EmitContext<CSharpEmitterOptions>) {
         enums: new Map(),
       },
     };
-    if (
-      context.program.diagnostics.length > 0 &&
-      context.program.diagnostics.filter((digs) => digs.severity === "error").length > 0
-    ) {
-      logDiagnostics(context.program.diagnostics, context.program.host.logSink);
-      process.exit(1);
-    }
+    program.reportDiagnostics(sdkContext.diagnostics);
```

Comment:
this invocation does not cause the program to exit when it contains errors.
Typespec compilation exits with error only when the invocation of an emitter is finished, in our case, if the compiler has other steps after http-client-csharp, those steps will not execute, but the code below will still execute normally.
I think this should be better, so that we could collect as many errors as possible to our customers.